File: Python/binary-watch.py
class Solution:
    def readBinaryWatch(self, num):
        """
        :type num: int
        :rtype: List[str]
        """
        res = list()

        def bitCount(num):
            count = 0
            while num:
                count += num & 1
                num >>= 1
            return count

        for h in range(12):
            for m in range(60):
                if bitCount(h) + bitCount(m) == num:
                    res.append('%d:%02d' % (h, m))

        return res

File: Swift/binary-watch.swift
class Solution {
    func readBinaryWatch(_ num: Int) -> [String] {
        var res = [String]()

        func bitCount(_ num: Int) -> Int {
            var count = 0
            var num = num
            while num > 0 {
                count += num & 1
                num >>= 1
            }
            return count
        }

        for h in 0...11 {
            for m in 0...59 {
                if bitCount(h) + bitCount(m) == num {
                    res.append(String(format: "%d:%02d", h, m))
                }
            }
        }

        return res
    }
}
